src/Symfony/Component/HttpKernel/EventListener/ValidationFailedExceptionListener.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Serializer\Exception\NotEncodableValueException;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Validator\Exception\ValidationFailedException;
+
+/**
+ * @author Konstantin Myakshin <molodchick@gmail.com>
+ */
+final class ValidationFailedExceptionListener implements EventSubscriberInterface
```

Comment:
Do we really need this? Looking at what we have already, `SerializerErrorRenderer` takes care of all this to me. We should figure out how to leverage it if it's not used yet.

Reply:
yes, we need that. `SerializerErrorRenderer` renders an exception but here we need to serialize `ConstraintViolationsList`

Reply:
I agree with Nicolas. You could create a new normalizer instead, which will handle the `FlattenException` and its inner exception, `ValidationFailedException`. Then, utilize the `ConstraintViolationListNormalizer` to normalize `$context['exception']->getViolations()`. For the http code, you could use a HTTP exception inside the value resolver, probably a new one that wraps also the violations.

Reply:
Moving exception handling into Argument Resolver will make impossible to handle `ValidationFailedException` which thrown outside of Argument Resolver (e.g. arbitrary services)